jaxrs/resteasy-jaxrs/src/main/java/org/jboss/resteasy/specimpl/ResteasyUriBuilder.java

```diff
@@ -656,7 +656,7 @@ protected StringBuilder replaceParameter(Map<String, ? extends Object> paramMap,
 				start = matcher.end();
 				continue;
 			}
-			throw new IllegalArgumentException(Messages.MESSAGES.pathParameterNotProvided(param));
+            throw new IllegalArgumentException(Messages.MESSAGES.templateParameterNull(param));

```

Comment:
Thanks for the review. I've swapped the 2 error messages, erroneously inverted them when chasing the spec/tck problems.